# Project 1: Deaths by tuberculosis

by Michel Wermelinger and Bashar, 16 December 2021, edited 5 April 2016

This is the project notebook for the first part of The Open University's _Learn to code for Data Analysis_ course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the BRICS (Brazil, Russia, India, China, South Africa) and Portuguese-speaking countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.1317?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [1]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
data = read_excel('WHO POP TB all.xls')
data

Country  Population (1000s)  TB deaths
0                           Afghanistan               30552   13000.00
1                               Albania                3173      20.00
2                               Algeria               39208    5100.00
3                               Andorra                  79       0.26
4                                Angola               21472    6900.00
..                                  ...                 ...        ...
189  Venezuela (Bolivarian Republic of)               30405     480.00
190                            Viet Nam               91680   17000.00
191                               Yemen               24407     990.00
192                              Zambia               14539    3600.00
193                            Zimbabwe               14150    5700.00

[194 rows x 3 columns]

## The range of the problem

The column of interest is the last one.

In [2]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [3]:
tbColumn.sum()

1072677.97

The largest and smallest number of deaths in a single country are:

In [4]:
tbColumn.max()

240000.0

In [5]:
tbColumn.min()

0.0

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [6]:
tbColumn.mean()

5529.267886597938

In [7]:
tbColumn.median()

315.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [8]:
data.sort_values('TB deaths')

Country  Population (1000s)  TB deaths
147  San Marino                  31       0.00
125        Niue                   1       0.01
111      Monaco                  38       0.03
3       Andorra                  79       0.26
129       Palau                  21       0.36
..          ...                 ...        ...
128    Pakistan              182143   49000.00
78    Indonesia              249866   64000.00
13   Bangladesh              156595   80000.00
124     Nigeria              173615  160000.00
77        India             1252140  240000.00

[194 rows x 3 columns]

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [9]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data

Country  Population (1000s)  TB deaths  \
0                           Afghanistan               30552   13000.00   
1                               Albania                3173      20.00   
2                               Algeria               39208    5100.00   
3                               Andorra                  79       0.26   
4                                Angola               21472    6900.00   
..                                  ...                 ...        ...   
189  Venezuela (Bolivarian Republic of)               30405     480.00   
190                            Viet Nam               91680   17000.00   
191                               Yemen               24407     990.00   
192                              Zambia               14539    3600.00   
193                            Zimbabwe               14150    5700.00   

     TB deaths (per 100,000)  
0                  42.550406  
1                   0.630318  
2                  13.007549  
3                   0.329114  
4                  32.134873  
..                       ...  
189                 1.578688  
190                18.542757  
191                 4.056213  
192                24.760988  
193                40.282686  

[194 rows x 4 columns]

In [27]:
data.sort_values('TB deaths (per 100,000)')

Country  Population (1000s)  TB deaths  TB deaths (per 100,000)
147   San Marino                  31       0.00                 0.000000
111       Monaco                  38       0.03                 0.078947
126       Norway                5043       4.40                 0.087250
120  Netherlands               16759      20.00                 0.119339
137        Qatar                2169       2.70                 0.124481
..           ...                 ...        ...                      ...
158      Somalia               10496    7700.00                73.361280
172  Timor-Leste                1133     990.00                87.378641
165    Swaziland                1250    1100.00                88.000000
124      Nigeria              173615  160000.00                92.157936
49      Djibouti                 873     870.00                99.656357

[194 rows x 4 columns]

In [17]:
data.shape

(194, 4)

In [16]:
data['TB deaths (per 100,000)'].argmax()

49

In [24]:
data['TB deaths'].describe()

count       194.000000
mean       5529.211340
std       22657.274105
min           0.000000
25%          25.750000
50%         315.000000
75%        2200.000000
max      240000.000000
Name: TB deaths, dtype: float64

## Conclusions

Due to TB, there are 194 countries which had a total of about one million and seven hundred and twinty seven thousands deaths. 

The median shows that half of these coutries had fewer than 315 deaths. The much higher mean (over 5529) indicates that some countries had a very high number. 

The least affected were San Marinio and Niue, with 0 and 0.01 deaths respectively, and the most affected were Nigeria and India with 160 thousand and 240 thousand deaths in a single year. However, taking the population size into account, the least affected were San Marino and Monaco with less than 1 death per 100 thousand inhabitants, and the most affected were Nigeria and Gjibouti with over 90 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.